### PR TITLE
bug 1491227. Modify request if user has back slash

### PR DESCRIPTION
**Pull request by jcantrill**
This PR fixes BZ1491227:
* Modifying the request header to convert backslash to forward
* Fixes role and role mapping generation to handle slashes

### CODE REVIEW

**Comment by jcantrill** [comment]
cc @portante

File: src/main/java/io/fabric8/elasticsearch/plugin/OpenshiftRequestContextFactory.java [hunk 1]
```diff
@@ -83,6 +83,10 @@ public OpenshiftRequestContext create(final RestRequest request, final UserProje
         Set<String> projects = new HashSet<>();
         boolean isClusterAdmin = false;
         String user = utils.getUser(request);
+        if(user.contains("\\")){
+            user = user.replaceAll("\\\\", "/");
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/OpenshiftRequestContextFactory.java]: We really need to escape this much? Is there a java util we can use to escape for us so this looks cleaner?

File: src/main/java/io/fabric8/elasticsearch/plugin/acl/BaseRolesSyncStrategy.java [hunk 2]
```diff
@@ -71,7 +71,7 @@ protected String formatKibanaRoleName(UserProjectCache cache, String username, S
     }
     
     public static String formatUserRoleName(String username) {
-        return String.format("%s_%s", USER_PREFIX, username.replaceAll("[\\.@]", "_"));
+        return String.format("%s_%s", USER_PREFIX, username.replaceAll("[\\\\.@/]", "_"));
```
Comment (ewolinetz) [inline src/main/java/io/fabric8/elasticsearch/plugin/acl/BaseRolesSyncStrategy.java]: Similar question about escaping so much

**Comment by ewolinetz** [comment]
Quick questions, otherwise LGTM

**Pull request merged**